### DISCUSSION TITLE
Integrating HARMONIZE climate data with Epiverse-TRACE tools

### DISCUSSION THREAD

**Discussion by adamkucharski** [epiverse-trace/epiverse-trace.github.io#79] [General]
Starting a discussion so we can share ideas for integration with the climate data tools coming out of [HARMONIZE](https://data.org/stories/harmonize/). Sounds like standardisation functionality for respective timeseries (e.g. comparing climate indicators with episoap outputs) would be useful starting point, but also scope for future integration with things like time-varying transmission in scenarios, epidemiological parameters, spatial layers etc.

**Comment by pratikunterwegs** [comment]
Great to get this started! I wonder whether something like [this use of social media data to forecast Covid in multiple geographic units](https://www.nature.com/articles/s41598-023-37118-9) could be done for climate data as well. For example, using weekly Sentinel thermal band data, or TROPOMI air quality data, to feed into epidemic models could be especially useful for regions where fine scale climate data are less available from national or local weather services. This could be a data-driven form of seasonal forcing (which we know @rozeggo especially likes).

Some great sources for large scale environmental and climate data are [Google's Earth Engine](https://developers.google.com/earth-engine/datasets/), and [Microsoft's Planetary Computer](https://planetarycomputer.microsoft.com/). Access via R is a bit janky (e.g. {rgee} for GEE works through Python's `gee` package using {reticulate}, iirc; aside: it's easier to use Python directly for such data using `gee` and the very excellent [`geemap` package](https://geemap.org/) whose docs are better than `gee` by far).

A lot of environmental climate data and access packages are reported in the ecological literature, and I would recommend Methods in Ecology and Evolution as a good port of first call to understand methods and datasets. They also publish software papers on R packages. See for example [this paper on meso- and micro-climate modelling](https://besjournals.onlinelibrary.wiley.com/doi/10.1111/2041-210X.13093) presenting the R package {microclima}, and this follow-up, very ambitiously titled [A method for computing hourly, historical, terrain-corrected microclimate anywhere on earth](https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.13330). See especially their conceptual workflow diagram that includes other useful R packages (Fig 1.). See also the R packages [{NicheMapr}](https://mrke.github.io/) and [{RangeShifter}](https://rangeshifter.github.io/) for species distribution-related modelling, could be useful for vector-borne disease modelling. The [CRAN task view on ecological and environmental packages](https://cran.r-project.org/web/views/Environmetrics.html) is another starting point.

**Comment by adamkucharski** [comment]
Added [issue](https://github.com/epiverse-trace/epidemics/issues/76) on first steps for integration of temperature varying transmission/parameters in epidemic models, which could link with data products above (HARMONIZE folks are going to advise more on where might be sensible starting point, as they will have a lot of the climate ensemble/modelling pipelines implemented within their project). More to come on {epiparameter} and other packages (e.g. around data cleaning/processing).

**Reply by pratikunterwegs** [reply]
Thanks - have just seen that, and will reply more in detail there. Would also be good to ping @rozeggo and @BlackEdder to point them to these discussions - I had just been proposing a vector-borne disease model after our CEPI call.

**Reply by adamkucharski** [reply]
Sounds good – happy to repurpose code from our earlier dengue/Zika work if useful to get an initial ODE model running (although probably best to wait for the vector parameters to be ready in {epiparameter} before including all the vector biology, rather than just a simple time-varying beta).

**Reply by pratikunterwegs** [reply]
Thanks - would be good to see previous code to look for general solutions.

**Reply by adamkucharski** [reply]
Our [Fiji model](https://github.com/adamkucharski/fiji-denv3-2014/blob/5cacdcd9e90b4a2e8fa6bd09b8550b8654f8ce06/R/dynamic_model_functions.R#L304) has the full vector parameters (plus interventions via `decline_f` and rainfall-dependent mosquito population via `carrying_f`, both of which could be excluded for an initial model). An easy way to simplify would be to turn off temperature dependency in the vector parameters, and make the beta terms time dependent:
```
beta_h1 <- mosquito_to_human(time)
beta_v1 <-  human_to_mosquito(time)
```
This would give a model structure more similar to the one in [Netto et al](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5686533/) for Zika. And with no time dependency at all, would get something like the [Funk et al analysis](https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0005173) of dengue/Zika in small island populations.

**Reply by pratikunterwegs** [reply]
Thanks, will link this in the issue as a reference.

For a status update, I'm still working on the ebola model (some trouble on Linux/Windows there) - what are we thinking of for a timeline for adding a vector-borne model with seasonality?

**Reply by adamkucharski** [reply]
It will take some time to get the temperature dependent functionality lined up in {epiparameter}, so exploring development of vector-borne model could wait until this is done if needed (as implementing the ODE with some additional compartments/rates will probably be quite quick, but wrangling the parameters/data-dependencies for temperature etc. will be tricker).

**Reply by pratikunterwegs** [reply]
Okay, sounds good - I'll pin the {epidemics} issue so we don't forget about it, and tentatively plan to check in on this again in mid-September or October. Happy to help with other parts of the Harmonize integration until then.
